lib/chef/knife/ec_base.rb

```diff
@@ -121,6 +127,33 @@ def rest
         @rest ||= Chef::ServerAPI.new(server.root_url, {:api_version => "0"})
       end
 
+      def remote_users
+        @remote_users ||= rest.get('/users')
+      end
+
+      def remote_user_list
+        @remote_user_list ||= remote_users.keys.map(&:to_sym)
```

Comment:
You're right - the speed savings aren't there.  I changed to an array of strings.